### PR TITLE
Added a Web Component for the web-player

### PR DESCRIPTION
**Pull request by SebastianLjungman**
Created a Web Component of the web-player
Also added webpack-dev-server for testing

### CODE REVIEW

File: packages/web-player-component/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@eyevinn/experimental",
```
Comment (slowmove) [inline packages/web-player-component/package.json:2]: This name should be aligned with what the package will be named at npm

File: packages/web-player-component/src/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import webplayer from '@eyevinn/web-player';
+import style from '@eyevinn/web-player/dist/webplayer.css';
+
+export default class PlayerComponent extends HTMLElement {
+  constructor() {
+    //Call constructor of HTMLElement
+    super();
+    //Create shadow root
+    this.attachShadow({ mode: 'open' });
+    const { shadowRoot } = this;
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:10]: We should discuss whether we should use a shadow dom at all vs render in the main document.

File: packages/web-player-component/src/index.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    shadowRoot.appendChild(styleTag);
+    //Create wrapper and attach to shadow DOM
+    const wrapper = document.createElement('div');
+    shadowRoot.appendChild(wrapper);
+    //Init player with wrapper
+    const player = webplayer(wrapper);
+    //Load video and play
+    player.load("https://storage.googleapis.com/shaka-demo-assets/sintel-mp4-only/dash.mpd").then(() => {
+      player.play()
+    });
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:23]: Let's rather take the source, as well as other options, as attributes on the HTML element.

File: packages/web-player-component/src/index.js [hunk 4]
```diff
@@ -15,22 +18,50 @@ export default class PlayerComponent extends HTMLElement {
...
+      if (this.getAttribute('muted') === 'true') {
+        this.video.muted = true;
+      }
+      else if (this.getAttribute('muted') === 'false') {
+        this.video.muted = false;
+      }
+    }
+    if (name === 'autoplay') {
+      if (this.getAttribute('autoplay') === 'true') {
+        this.video.setAttributeNode(document.createAttribute('autoplay'));
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:49]: isn't it enough to just use `setAttribute`?

File: packages/web-player-component/src/index.js [hunk 5]
```diff
@@ -15,22 +18,50 @@ export default class PlayerComponent extends HTMLElement {
...
+      }
+      else if (this.getAttribute('autoplay') === 'false') {
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:51]: Shouldn't we look for all falsy scenarios here, not just `false`?

File: packages/web-player-component/src/index.js [hunk 6]
```diff
@@ -15,22 +18,50 @@ export default class PlayerComponent extends HTMLElement {
...
+  attributeChangedCallback(name) {
+    console.log(name);
+    if (name === 'source') {
+      this.player.load(this.getAttribute('source'));
+    }
+    if (name === 'muted') {
+      if (this.getAttribute('muted') === 'true') {
+        this.video.muted = true;
+      }
+      else if (this.getAttribute('muted') === 'false') {
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:43]: Shouldn't we look for all falsy scenarios here, not just `false`?

File: packages/web-player-component/src/index.js (hunk 4 above)
Comment (SebastianLjungman) [reply packages/web-player-component/src/index.js:49]: This works if we set 'autoplay'='autoplay' (can't be true/false)

File: packages/web-player-component/src/index.js (hunk 5 above)
Comment (SebastianLjungman) [reply packages/web-player-component/src/index.js:51]: Replace with **else**?

File: packages/web-player-component/src/index.js (hunk 6 above)
Comment (SebastianLjungman) [reply packages/web-player-component/src/index.js:43]: Replace with **else**?

File: packages/web-player-component/package.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@eyevinn/web-player-component",
+  "version": "1.0.0",
+  "description": "> TODO: description",
+  "author": "SebastianLjungman <sebastian.ljungman94@gmail.com>",
+  "homepage": "https://github.com/SebastianLjungman/web-player#readme",
```
Comment (birme) [inline packages/web-player-component/package.json:6]: Replace this with https://web.player.eyevinn.technology/ instead

File: packages/web-player-component/package.json [hunk 8]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@eyevinn/web-player-component",
+  "version": "1.0.0",
+  "description": "> TODO: description",
+  "author": "SebastianLjungman <sebastian.ljungman94@gmail.com>",
```
Comment (birme) [inline packages/web-player-component/package.json:5]: Change author to be `Eyevinn Technology <work@eyevinn.se>` Add your personal info to contributions section instead. Please also use the Eyevinn email address

File: packages/web-player-component/package.json [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  "directories": {
+    "lib": "lib",
+    "test": "__tests__"
+  },
+  "files": [
+    "lib"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/SebastianLjungman/web-player.git"
```
Comment (birme) [inline packages/web-player-component/package.json:18]: This should point to the web player repository

File: packages/web-player-component/package.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  "scripts": {
+    "build": "rm -rf dist/ && webpack --config webpack.config.js",
+    "dev": "webpack-dev-server --hot"
+  },
+  "bugs": {
+    "url": "https://github.com/SebastianLjungman/web-player/issues"
```
Comment (birme) [inline packages/web-player-component/package.json:25]: Point to the eyevinn web-player repo

File: packages/web-player-component/src/index.html [hunk 11]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <title>Web Player Component Demo</title>
+    </head>
+    <body>
+        <!-- Browsers will block autoplay if muted is false -->
+        <player-component source="https://maitv-vod.lab.eyevinn.technology/VINN.mp4/master.m3u8" starttime="25" muted autoplay></player-component>
```
Comment (birme) [inline packages/web-player-component/src/index.html:10]: I propose we add the eyevinn name in the component, suggesting:
`<eyevinn-video>`

File: packages/web-player-component/package.json [hunk 12]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@eyevinn/web-player-component",
+  "version": "1.0.0",
+  "description": "> TODO: description",
+  "author": "SebastianLjungman <sebastian.ljungman94@gmail.com>",
+  "homepage": "https://github.com/SebastianLjungman/web-player#readme",
+  "license": "ISC",
+  "main": "lib/experimental.js",
```
Comment (birme) [inline packages/web-player-component/package.json:8]: change this, right?

**Comment by SebastianLjungman** [comment]
Kept working in branch other branch "addEmbedButton"

**Pull request closed without merging**